### PR TITLE
Add PMC E2E tests

### PR DESCRIPTION
**Pull request by Salazander**
## Description

Adds end-to-end tests for the PMC module. Contains both fair-weather and error scenarios.

**IMPORTANT**: Environment variable `E2E_OSCONFIG_DISTRIBUTION_NAME` needs to be set to a valid supported Debian distro name: `["ubuntu-18.04", "ubuntu-20.04", "debian9"]`. Tests will be ignored if this environment variable is not set on the executing system (e.g. CI agent). The targeted edge device needs the following tools as prerequisite: `["apt-get", "apt-cache", "dpkg-query", "curl", "gpg"]`

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: devops/pipeline/stages/e2etest_baremetal.yaml [hunk 1]
```diff
@@ -135,4 +136,5 @@ stages:
       deviceStageName: Baremetal_${{ parameters.deviceId }}
       testExpression: 'TestCategory != BaremetalExclude'
       twinTimeout: ${{ parameters.twinTimeout }}
-      environment: ${{ parameters.environment }}
\ No newline at end of file
+      environment: ${{ parameters.environment }}
+      distroName: debian9
```
Comment (AhmedBM) [inline devops/pipeline/stages/e2etest_baremetal.yaml:140]: The hardcoded `debian9` is fine for now, we are changing all our targets to be generic going forward, no more cloud/baremetal

**Review by AhmedBM — APPROVED** [review]
Looks good!

**Comment by AhmedBM** [comment]
I was able to validate the changes on my end and it looks good! 👌

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**